core/src/main/java/org/apache/struts2/dispatcher/multipart/JakartaMultiPartRequest.java

```diff
@@ -186,13 +191,14 @@ protected ServletFileUpload createServletFileUpload(DiskFileItemFactory fac) {
     }
 
     protected DiskFileItemFactory createDiskFileItemFactory(String saveDir) {
-        DiskFileItemFactory fac = new DiskFileItemFactory();
+        DiskFileItemFactory.Builder fac = DiskFileItemFactory.builder();
         // Make sure that the data is written to file, even if the file is empty.
-        fac.setSizeThreshold(-1);
+        //setting 0 or -1 no longer seems to work for fileupload buffer size, so using 1 instead.
+        fac.setBufferSize(1);
         if (saveDir != null) {
-            fac.setRepository(new File(saveDir));
+            fac.setPath(saveDir);
```

Comment:
@jdyer1 so the test is failing here. 

i did make a mistake, this should be` fac.setPath(new File(saveDir))` i believe. But even when doing that, you're still not able to get the path back at all. 

This seems like a bug with the M1 release of fileupload2, I'm never able to get a path back for files made with the DiskFileItemFactory, but i could be overlooking something too.

Reply:
Maybe this is a question for the commons-upload dev list?  I wonder, also, is it vital that Struts be able to tell the difference between a missing file and an empty file?  This is the only problem we are having here, correct?